### PR TITLE
DR-1369 Fix exception handling for firestore batches

### PR DESCRIPTION
**Pull request by ddietterich**
I also added logging properties that may or may not do anything to get dbcp2 to tell us what is going on.

### CODE REVIEW

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java [hunk 1]
```diff
@@ -217,4 +216,17 @@ String computeCrc32c(String input) {
...
+            return false; // Did not find a retry-able exception
+        }
+        if (throwable instanceof DeadlineExceededException ||
+            throwable instanceof UnavailableException) {
+            logger.warn("Retry-able error in firestore future get - input: " +
+                input + " message: " + throwable.getMessage());
+            return true;
+        }
+        return shouldRetry(throwable.getCause(), input);
+    }
```
Comment (RoriCremer) [inline src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java:230]: What's the over under that we're gonna find a third one?

**Review by RoriCremer — APPROVED** [review]

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java (hunk 1 above)
Comment (ddietterich) [reply src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java:230]: I don't want to be the one to find out :)

File: src/main/java/bio/terra/app/configuration/JdbcConfiguration.java [hunk 2]
```diff
@@ -99,6 +99,8 @@ private void configureDataSource() {
         props.setProperty("password", getPassword());
         props.setProperty("maxTotal", String.valueOf(poolMaxTotal));
         props.setProperty("maxIdle", String.valueOf(poolMaxIdle));
+        props.setProperty("logExpiredConnections", "true");
```
Comment (nmalfroy) [inline src/main/java/bio/terra/app/configuration/JdbcConfiguration.java:106]: It this a temporary thing or do we want to add that to be configurable. E.g. should these be member variables in this class?

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java [hunk 3]
```diff
@@ -217,4 +216,17 @@ String computeCrc32c(String input) {
         return outputs;
     }
 
+    private <V> boolean shouldRetry(Throwable throwable, V input) {
```
Comment (nmalfroy) [inline src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java:219]: I don't think it's part of this commit but could this get stuck in a loop?  e.g. is there an upper limit to the number of retries?

**Review by nmalfroy — APPROVED** [review]
Looks good in order to have Raaid do a test run.  Couple of comments/questions but those don't need to hold up a merge

File: src/main/java/bio/terra/app/configuration/JdbcConfiguration.java (hunk 2 above)
Comment (ddietterich) [reply src/main/java/bio/terra/app/configuration/JdbcConfiguration.java:106]: I intend it to be temporary to debug the current problem. I'll put a comment in to make that clear.

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java (hunk 3 above)
Comment (ddietterich) [reply src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java:219]: The batching code has a check: if it doesn't make any progress after `NO_PROGRESS_MAX` retries, it gives up. That is currently set to 2.

Comment (nmalfroy) [reply src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java:219]: Got it, thanks!

File: src/main/java/bio/terra/app/configuration/JdbcConfiguration.java (hunk 2 above)
Comment (nmalfroy) [reply src/main/java/bio/terra/app/configuration/JdbcConfiguration.java:106]: Ok, thought maybe but just wanted to check

**Pull request merged**